pkg/api/common.go

```diff
@@ -26,10 +29,74 @@ var (
+	flag.BoolVar(&cfg.ReadOnly, "read-only", false, "Read-only mode for the connector. Operations related to writing or updating the database are disallowed. It is used when pointing the connector to a TimescaleDB read replica.")
+	flag.BoolVar(&cfg.AdminAPIEnabled, "web-enable-admin-api", false, "Allow operations via API that are for advanced users. Currently, these operations are limited to deletion of series.")
+
+	flag.StringVar(&cfg.BasicAuthUsername, "basic-auth-username", "", "HTTP Basic Authentication Username used for web endpoint authentication. Disabled by default.")
+	flag.StringVar(&cfg.BasicAuthPassword, "basic-auth-password", "", "HTTP Basic Authentication Password used for web endpoint authentication. Password or password-file option should be set together with username.")
+	flag.StringVar(&cfg.BasicAuthPasswordFile, "basic-auth-password-file", "", "HTTP Basic Authentication Password file containing the actuall password used for web endpoint authentication. Password or password-file option should be set together with username.")
+	flag.StringVar(&cfg.BearerToken, "bearer-token", "", "Bearer token (JWT) used for web endpoint authentication. Disabled by default. Cannot be used together with bearer token file setting.")
+	flag.StringVar(&cfg.BearerTokenFile, "bearer-token-file", "", "File path containing the bearer token (JWT) used for web endpoint authentication. Disabled by default. Cannot be used together with bearer token setting.")
+	return cfg
+}
+
+func ValidateConfig(cfg *Config) error {
+	switch {
+	case cfg.BasicAuthUsername != "":
+		if cfg.BearerToken != "" || cfg.BearerTokenFile != "" {
+			return fmt.Errorf("at most one of basic-auth-username, bearer-token & bearer-token-file must be set")
+		}
+		if cfg.BasicAuthPassword == "" && cfg.BasicAuthPasswordFile == "" {
+			return fmt.Errorf("one of basic-auth-password & basic-auth-password-file must be configured")
+		}
+		if cfg.BasicAuthPassword != "" && cfg.BasicAuthPasswordFile != "" {
+			return fmt.Errorf("at most one of basic-auth-password & basic-auth-password-file must be configured")
+		}
+		pwd, err := readFromFile(cfg.BasicAuthPasswordFile, cfg.BasicAuthPassword)
+		if err != nil {
+			return fmt.Errorf("error reading password file: %w", err)
+		}
+		cfg.BasicAuthPassword = pwd
+	case cfg.BearerToken != "":
+		if cfg.BearerTokenFile != "" {
+			return fmt.Errorf("at most one of bearer-token & bearer-token-file must be set")
+		}
+	case cfg.BearerTokenFile != "":
+		token, err := readFromFile(cfg.BearerTokenFile, "")
+		if err != nil {
+			return fmt.Errorf("error reading bearer token file: %w", err)
+		}
+		cfg.BearerToken = token
+	}
+
+	return nil
+}
+
+func readFromFile(path string, defaultValue string) (string, error) {
+	if path == "" {
+		return defaultValue, nil
+	}
+	bs, err := ioutil.ReadFile(path)
+
+	if err != nil {
+		return "", fmt.Errorf("unable to read file %s: %w", path, err)
+	}
+
+	return strings.TrimSpace(string(bs)), nil
```

Comment:
I don't know of an authentication method that should preserver trailing spaces. Maybe bearer token?

Reply:
Do we really need to trim space?

Reply:
I don't see an issue with trimming space. The only thing we can do there is fix file contents if somebody were to store some unintentional whitespace in the file. 

I guess its possible to use whitespace in a password, in that case this could cause an issue.

Reply:
>  don't see an issue with trimming space

Why do we need to trim space? Is it that secrets lead to unintended padding while marshalling? 
I am fine with trimming but there has to be a need of it which I do not know. Can you please mention the reason of trimming.

Reply:
> The only thing we can do there is fix file contents if somebody were to store some unintentional whitespace in the file.

That is the reason. You can see similar functionality in other projects.

Reply:
Ah, got it. We can go ahead this way but I still think that if someone is having white spaces, then its his/her fault and we doing this silently is not good.